plugins/BEdita/Core/src/Model/Action/CloneAction.php

```diff
@@ -0,0 +1,251 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2024 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Model\Action;
+
+use BEdita\Core\Model\Entity\ObjectEntity;
+use BEdita\Core\Model\Entity\Stream;
+use BEdita\Core\Utility\LoggedUser;
+use BEdita\Core\Utility\Schema;
+use BEdita\Core\Utility\Text;
+use Cake\Datasource\EntityInterface;
+use Cake\ORM\Locator\LocatorAwareTrait;
+use Cake\Utility\Hash;
+
+/**
+ * Clone action
+ */
+class CloneAction extends BaseAction
+{
+    use LocatorAwareTrait;
+
+    /**
+     * Table.
+     *
+     * @var \Cake\ORM\Table
+     */
+    protected $Table;
+
+    /**
+     * Author ID.
+     *
+     * @var int
+     */
+    protected $authorId;
+
+    /**
+     * @inheritDoc
+     */
+    protected function initialize(array $data)
+    {
+        $this->Table = $this->getConfig('table');
+        $this->authorId = empty(LoggedUser::id()) ? LoggedUser::getUserAdmin()['id'] : LoggedUser::id();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(array $data = [])
+    {
+        $sourceId = (int)Hash::get($data, 'id');
+        $include = (array)Hash::get($data, 'data._meta.include');
+        $attributes = array_filter($data['data'], function ($key) {
+            return !in_array($key, ['_meta']);
+        }, ARRAY_FILTER_USE_KEY);
+        $entity = null;
+        $this->Table->getConnection()->transactional(function () use (&$entity, $sourceId, $attributes, $include) {
+            $objectType = $this->Table->objectType();
+            $options = $objectType->hasAssoc('Streams') ? ['contain' => ['Streams']] : [];
```

Comment:
What about other possible associations as `DateRanges`, `Captions`, `Categories`, etc... . Maybe we can find a way to clone those associations reading from `ObjectType::$associations` and execute operations based on the association type.